src/pages/mypage/pages/user-profile/index.tsx

```diff
@@ -1,3 +1,136 @@
+import { useNavigate } from 'react-router-dom';
+import {
+  useGetUsersMeQuery,
+  usePatchUsersMeMutation,
+  usePostUsersCheckNicknameMutation,
+} from '@/queries/users';
+import { useState } from 'react';
+import {
+  fixBox,
+  fixContainer,
+  fixInfo,
+  line,
+  titleContainer,
+  fixedButtonWrapper,
+  placeholderStyle,
+} from './index.styles';
+import theme from '@/styles/theme';
+
 export default function UserProfile() {
-  return <></>;
+  const navigate = useNavigate();
+  const { data } = useGetUsersMeQuery();
+
+  const [nickname, setNickname] = useState(data?.nickname || '');
+  const [phone, setPhone] = useState(data?.phone || '');
+  const [bankAccount, setBankAccount] = useState(
+    data?.bankAccount ? `${data.bankAccount.bank} ${data.bankAccount.accountNumber}` : ''
+  );
+  const [isNicknameAvailable, setIsNicknameAvailable] = useState<boolean | null>(null);
+
+  const { mutate: patchUsersMe } = usePatchUsersMeMutation(() => {
+    alert('프로필이 성공적으로 수정되었습니다.');
+    navigate('/');
+  });
+
+  const { mutate: checkNicknameMutation } = usePostUsersCheckNicknameMutation(() => {
+    setIsNicknameAvailable(true);
+  });
+
+  const checkNickname = () => {
+    checkNicknameMutation(
+      { nickname },
+      {
+        onError: () => {
+          setIsNicknameAvailable(false);
+        },
+      }
+    );
+  };
+
+  const handleSubmit = () => {
+    const [bank, accountNumber] = bankAccount.split(' ');
+    patchUsersMe({ nickname, phone, bankAccount: { bank, accountNumber } });
+  };
+
+  return (
+    <Layout>
+      <AppBar
+        prefix={
+          <AppBarBack
+            height="17px"
+            cursor="pointer"
+            onClick={() => {
+              navigate(-1);
+            }}
+          />
+        }
+      />
+      <div css={titleContainer}>
+        <Text typo="title1" color={theme.colors.black}>
+          프로필 수정
+        </Text>
+      </div>
+      <hr css={line} />
+      <div css={fixContainer}>
+        <div css={fixBox}>
+          <Text typo="subtitle2" color={theme.colors.black}>
+            이름
+          </Text>
+          <div css={fixInfo}>
+            <Text typo="subtitle1" color={theme.colors.gray300}>
+              {data?.username}
+            </Text>
+          </div>
+        </div>
+        <div css={fixBox}>
+          <Text typo="subtitle2" color={theme.colors.black}>
+            이메일
+          </Text>
+          <div css={fixInfo}>
+            <Text typo="subtitle1" color={theme.colors.gray300}>
+              {data?.email}
+            </Text>
+          </div>
+        </div>
+        <Input
+          type="text"
+          label="닉네임"
+          placeholder="닉네임을 입력해주세요"
+          css={placeholderStyle}
+          value={nickname}
+          onChange={(e) => setNickname(e.target.value)}
+          confirmMessage={isNicknameAvailable === true ? '사용 가능한 닉네임입니다' : ''}
+          errorMessage={isNicknameAvailable === false ? '이미 존재하는 닉네임입니다' : ''}
```

Comment:
빈 문자열을 넘기는 것보다는 이렇게 하는 게 더 좋을 것 같아요
```suggestion
          confirmMessage={isNicknameAvailable && '사용 가능한 닉네임입니다'}
          errorMessage={!isNicknameAvailable && '이미 존재하는 닉네임입니다'}
```

Reply:
그리고 닉네임 형식에 맞는지도 검증 로직을 추가해야 할 것 같습니다
e.g. 2자 이상 10자 이하 입력